### PR TITLE
massively reduce build time 

### PR DESCRIPTION
**Pull request by cbielow**
# Description

... for class tests by removing includes from ClassTest.h.

Stats for Windows (which is the slowest platform ATM)

Build time for all class tests (16 threads):
before: 38min 
after: 26min (-32%)

Lets see what happens with CI.

Runtime was not negatively affected.

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

### CODE REVIEW

File: src/openms/include/OpenMS/METADATA/MetaInfoInterfaceUtils.h [hunk 1]
```diff
@@ -107,9 +107,9 @@ namespace OpenMS
         }
       }
       // pick the keys which occur often enough
-      const uint required_counts = uint(min_frequency / 100.0 * std::distance(it_start, it_end));
+      const UInt required_counts = UInt(min_frequency / 100.0 * std::distance(it_start, it_end));
       T_Out common_keys;
-      for (std::map<String, uint>::const_iterator it = counter.begin(); it != counter.end(); ++it)
+      for (auto it = counter.begin(); it != counter.end(); ++it)
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/METADATA/MetaInfoInterfaceUtils.h:112]: perfect for structure binding

File: src/openms/source/CONCEPT/ClassTest.cpp [hunk 2]
```diff
@@ -94,6 +96,133 @@ namespace OpenMS
     namespace ClassTest
     {
 
+      void mainInit(const char* version, const char* class_name, int argc, const char* argv0)
+      {
+      
+      OpenMS::UInt64 seed = 2453440375;
```
Comment (timosachsenberg) [inline src/openms/source/CONCEPT/ClassTest.cpp:102]: indentation? const?

File: src/openms/include/OpenMS/CONCEPT/ClassTest.h [hunk 3]
```diff
@@ -183,6 +175,13 @@ namespace OpenMS
       void OPENMS_DLLAPI
       printWithPrefix(const std::string& text, const int marked = -1);
 
+      /// init main stuff
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/CONCEPT/ClassTest.h:178]: if you know that these methods do can you add a comment? ideally with @param

**Comment by jpfeuffer** [comment]
Worked:
Mac 51 -> 35
Lnx 68 -> 51
Win not ready yet

**Comment by cbielow** [comment]
thanks for gathering the stats and whoohoo :)

**Review by timosachsenberg — APPROVED** [review]

**Pull request merged**